File: src/neetcode/Java/_3LongestSubstringWithoutRepeatingCharacters.java
package neetcode.Java;

public class _3LongestSubstringWithoutRepeatingCharacters {
    public static int lengthOfLongestSubstring(String s) {
        if (s.length() <= 1) return s.length();
        int longLength = 1, i = 0, n = 0, j;
        for (; i < s.length() - 1; i++, n = i + 1) {
            Task:
            for (; n < s.length(); ) {

                for (j = i; j < n; j++) {

                    if (s.charAt(j) == (s.charAt(n))) {
                        longLength = Math.max(longLength, n - i);
                        n = n + 1;
                        break Task;
                    }
                }
                if (j == n) {
                    n++;
                    longLength = Math.max(longLength, n - i);
                }
            }
            if (n == s.length()) return longLength;
        }
        return (longLength);
    }

    public static void main(String[] args) {
        int res = lengthOfLongestSubstring("abcabcbb");
        System.out.println(res);
    }
}

File: src/neetcode/kotlin/_3LongestSubstringWithoutRepeatingCharacters.kt
package neetcode.kotlin

import kotlin.math.max

object _3LongestSubstringWithoutRepeatingCharacters {
    fun lengthOfLongestSubstring(s: String): Int {
        if (s.length <= 1) return s.length
        var longLength = 1
        var i = 0
        var n = 0
        var j: Int
        while (i < s.length - 1) {
            Task@ while (n < s.length) {
                j = i
                while (j < n) {
                    if (s[j] == (s[n])) {
                        longLength = max(longLength.toDouble(), (n - i).toDouble()).toInt()
                        n = n + 1
                        break@Task
                    }
                    j++
                }
                if (j == n) {
                    n++
                    longLength = max(longLength.toDouble(), (n - i).toDouble()).toInt()
                }
            }
            if (n == s.length) return longLength
            i++
            n = i + 1
        }
        return (longLength)
    }

    @JvmStatic
    fun main(args: Array<String>) {
        val res = lengthOfLongestSubstring("abcabcbb")
        println(res)
    }
}
